lib/Sema/TypeCheckExprObjC.cpp

```diff
@@ -215,8 +215,16 @@ Optional<Type> TypeChecker::checkObjCKeyPathExpr(DeclContext *dc,
                diag::expr_unsupported_objc_key_path_component,
                (unsigned)kind);
       continue;
-    case KeyPathExpr::Component::Kind::OptionalWrap:
     case KeyPathExpr::Component::Kind::Property:
+    case KeyPathExpr::Component::Kind::Type: {
+      //For code completion purposes, we only care about the last expression.
+      if (&component == &expr->getComponents().back()) {
+        return component.getComponentType();
+      } else {
+        continue;
```

Comment:
This is also changing the behavior for `Kind::Property`, is that intentional?

Reply:
That's because you could in theory be reaching the keypath property from other properties too, like in `#keyPath(MyType.myProperty.hashValue)`.